### PR TITLE
[3006.x] bump pygit2 in ci

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
bump pygit2 dependence in ci
 
### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/65085

### Previous Behavior
py37 has unsupported version

### New Behavior
pygit2 will have less crashes and leaks and py 37 wont have problems 

### Merge requirements satisfied?
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html

### Commits signed with GPG?
Yes

### CODE REVIEW

File: requirements/static/ci/darwin.in [hunk 1]
```diff
@@ -3,7 +3,8 @@
 # pylxd(or likely ws4py) will cause the test suite to hang at the finish line under runtests.py
 # pylxd>=2.2.5
 yamlordereddictloader
-pygit2>=1.2.0
+pygit2>=1.10.1; python_version >= '3.7'
+pygit2>=1.13.1; python_version >= '3.8'
```
Comment (s0undt3ch) [inline requirements/static/ci/darwin.in:7]:
```suggestion
pygit2>=1.10.1
```
Then, once this change lands on the `master` branch we update to:
```
pygit2>=1.13.1
```

File: requirements/static/ci/freebsd.in [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # FreeBSD static CI requirements
-pygit2==1.8.0
+pygit2>=1.10.1; python_version >= '3.7'
+pygit2>=1.13.1; python_version >= '3.8'
```
Comment (s0undt3ch) [inline requirements/static/ci/freebsd.in:3]:
```suggestion
pygit2>=1.10.1
```
Then, once this change lands on the `master` branch we update to:
```
pygit2>=1.13.1
```

File: requirements/static/ci/linux.in [hunk 3]
```diff
@@ -1,8 +1,7 @@
 # Linux static CI requirements
 pyiface
-pygit2<1.1.0; python_version <= '3.8'
-pygit2>=1.4.0; python_version > '3.8'
-pygit2==1.9.1; python_version >= '3.10'
+pygit2>=1.10.1; python_version >= '3.7'
+pygit2>=1.13.1; python_version >= '3.8'
```
Comment (s0undt3ch) [inline requirements/static/ci/linux.in:4]:
```suggestion
pygit2>=1.10.1
```
Then, once this change lands on the `master` branch we update to:
```
pygit2>=1.13.1
```

File: requirements/static/ci/windows.in [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # This is a compilation of requirements installed on salt-jenkins git.salt state run
 dmidecode
 patch
-pygit2>=1.2.0
+pygit2>=1.10.1; python_version >= '3.7'
+pygit2>=1.13.1; python_version >= '3.8'
```
Comment (s0undt3ch) [inline requirements/static/ci/windows.in:5]:
```suggestion
pygit2>=1.10.1
```
Then, once this change lands on the `master` branch we update to:
```
pygit2>=1.13.1
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by cmcmarrow** [comment]
closing for https://github.com/saltstack/salt/pull/65430

**Pull request closed without merging**